src/bitcoin-tx.cpp

```diff
@@ -240,10 +240,10 @@ static void MutateTxAddInput(CMutableTransaction& tx, const std::string& strInpu
         throw std::runtime_error("TX input missing separator");
 
     // extract and validate TXID
-    std::string strTxid = vStrInputParts[0];
-    if ((strTxid.size() != 64) || !IsHex(strTxid))
-        throw std::runtime_error("invalid TX input txid");
-    uint256 txid(uint256S(strTxid));
+    uint256 txid;
+    if (!ParseHashStr(vStrInputParts[0], txid)) {
+        throw std::runtime_error("txid must be hexadecimal string (not '" + vStrInputParts[0] + "')");
```

Comment:
I know this is unchanged from the previous version, but could there be circumstances where this opens up for log file injection? Safer not to echo the input?